Python/Tests/DebuggerUITests/DebugProjectUITests.cs

```diff
@@ -40,202 +41,263 @@ public class DebugProjectUITests {
+            using (new PythonExperimentalGeneralOptionsSetter(pyService, useVsCodeDebugger: useVsCodeDebugger)) {
+                if (useVsCodeDebugger) {
+                    KillPythonProcess(TimeSpan.FromSeconds(30));
+                }
 
-            using (new EnvironmentVariableSetter("PYTHONPATH", testDataPath)) {
-                app.OpenProject(sysPathSln);
+                var sysPathSln = app.CopyProjectForTest(@"TestData\SysPath.sln");
+                var helloWorldSln = app.CopyProjectForTest(@"TestData\HelloWorld.sln");
+                var testDataPath = Path.Combine(PathUtils.GetParent(helloWorldSln), "HelloWorld").Replace("\\", "\\\\");
+
+                using (new EnvironmentVariableSetter("PYTHONPATH", testDataPath)) {
+                    app.OpenProject(sysPathSln);
+
+                    using (new PythonServiceGeneralOptionsSetter(pyService, clearGlobalPythonPath: false)) {
+                        ClearOutputWindowDebugPaneText(app);
+                        app.Dte.ExecuteCommand("Debug.Start");
+                        WaitForMode(app, dbgDebugMode.dbgDesignMode);
+
+                        WaitForDebugOutput(app, text => text.Contains(testDataPath));
+                    }
 
-                using (new PythonServiceGeneralOptionsSetter(pyService, clearGlobalPythonPath: false)) {
                     ClearOutputWindowDebugPaneText(app);
                     app.Dte.ExecuteCommand("Debug.Start");
                     WaitForMode(app, dbgDebugMode.dbgDesignMode);
 
-                    WaitForDebugOutput(app, text => text.Contains(testDataPath));
+                    var outputWindowText = WaitForDebugOutput(app, text => text.Contains("DONE"));
+                    Assert.IsFalse(outputWindowText.Contains(testDataPath), outputWindowText);
                 }
-
-                ClearOutputWindowDebugPaneText(app);
-                app.Dte.ExecuteCommand("Debug.Start");
-                WaitForMode(app, dbgDebugMode.dbgDesignMode);
-
-                var outputWindowText = WaitForDebugOutput(app, text => text.Contains("DONE"));
-                Assert.IsFalse(outputWindowText.Contains(testDataPath), outputWindowText);
             }
         }
 
         /// <summary>
         /// Tests using a custom interpreter path that is relative
         /// </summary>
-        public void DebugPythonCustomInterpreter(VisualStudioApp app, DotNotWaitOnNormalExit optionSetter) {
-            var sln = app.CopyProjectForTest(@"TestData\RelativeInterpreterPath.sln");
-            var project = app.OpenProject(sln, "Program.py");
-            var interpreterPath = Path.Combine(PathUtils.GetParent(sln), "Interpreter.exe");
-            File.Copy(PythonPaths.Python27.InterpreterPath, interpreterPath, true);
+        public void DebugPythonCustomInterpreter(VisualStudioApp app, bool useVsCodeDebugger, DotNotWaitOnNormalExit optionSetter) {
```

Comment:
Is the `optionSetter` argument actually used anywhere? It doesn't look like it was, even before the change. Ditto in other methods here.

Maybe get rid of that class altogether, if this was the only place where it was used?

Reply:
Test runner inspects the method parameters, creates an instance of that class and passes it in. After execution, it casts to IDisposable and calls Dispose on it. So it's used, but you don't see the code.